doc/datasets/index.rst

```diff
@@ -9,49 +9,61 @@ Dataset loading utilities
-used by the machine learning community to benchmark algorithm on data
-that comes from the 'real world'.
-
 General dataset API
 ===================
 
-There are three distinct kinds of dataset interfaces for different types
-of datasets.
-The simplest one is the interface for sample images, which is described
-below in the :ref:`sample_images` section.
+There are three main kinds of dataset interfaces that can be used to get 
+datasets depending on the desired type of dataset.
+  
+**The dataset loaders.** They can be used to load small standard datasets, 
+described in the :ref:`toy_datasets` section.  
 
-The dataset generation functions and the svmlight loader share a simplistic
-interface, returning a tuple ``(X, y)`` consisting of a ``n_samples`` *
+**The dataset fetchers.** They can be used to download and load larger datasets,
+described in the :ref:`real_world_datasets` section.
+
+Both loaders and fetchers functions return a dictionary-like object holding 
+at least two items: an array of shape ``n_samples`` * ``n_features`` with 
+key ``data``(except for 20newsgroups) and a numpy array of 
+length ``n_samples``, containing the target values, with key ``target``.
+
+It's also possible for almost all of these function to constrain the output
+to be a tuple containing only the data and the target, by setting the 
+``return_X_y`` parameter to ``True``.
+
+The datasets also contain a full description in their ``DESCR`` attribute and 
+some contain ``feature_names`` and ``target_names``. See the dataset 
+descriptions below for details.  
+
+**The dataset generation functions.** They can be used to generate controled 
+synthetic datasets, described in the :ref:`generated_datasets` section.
+
+These functions return a tuple ``(X, y)`` consisting of a ``n_samples`` *
 ``n_features`` numpy array ``X`` and an array of length ``n_samples``
 containing the targets ``y``.
 
-The toy datasets as well as the 'real world' datasets and the datasets
-fetched from mldata.org have more sophisticated structure.
-These functions return a dictionary-like object holding at least two items:
-an array of shape ``n_samples`` * ``n_features`` with key ``data``
-(except for 20newsgroups)
-and a numpy array of length ``n_samples``, containing the target values,
-with key ``target``.
-
-The datasets also contain a description in ``DESCR`` and some contain
-``feature_names`` and ``target_names``.
-See the dataset descriptions below for details.
+In addition, there are also miscellanous tools to load datasets of other 
+formats or from other locations, described in the :ref:`loading_other_datasets`
+section. 
 
 .. _toy_datasets:
 
 Toy datasets
 ============
 
-scikit-learn comes with a few small standard datasets that do not
-require to download any file from some external website. 
+scikit-learn comes with a few small standard datasets that do not require to 
+download any file from some external website. A description of each dataset is
+available below.
```

Comment:
"A description of each dataset is available below." could probably be trimmed to keep the documentation as informative as possible.